internal/kubernetes/client.go

```diff
@@ -0,0 +1,154 @@
+	var result []Ingress
+
+	log, err := c.kubeRunner.Get("all", "ingress", "-o",
+		"jsonpath='[{range .items[*]}{\"{\"}\"Name\": \"{.metadata.name}\", "+
+			"\"Host\": [{range .spec.rules[*]}\"{.host}\",{end}]{\"}\"},{end}]'")
+	if err != nil {
+		return result, fmt.Errorf("error while reading resources from cluster: %w", err)
+	}
+
+	reg := regexp.MustCompile(`'(.*?)'`)
+
+	logStringIndex := reg.FindStringIndex(log)
+
+	if logStringIndex == nil {
+		return result, nil
+	}
+
+	out := log[logStringIndex[0]+1 : logStringIndex[1]-1]
+
+	out = strings.ReplaceAll(out, ",]", "]")
+
+	err = json.Unmarshal([]byte(out), &result)
+	if err != nil {
+		return result, fmt.Errorf("error while unmarshaling json: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Client) GetPersistentVolumes() ([]string, error) {
+	log, err := c.kubeRunner.Get("all", "pv", "-o", "jsonpath='{.items[*].metadata.name}'")
+	if err != nil {
+		return []string{}, fmt.Errorf("error while reading resources from cluster: %w", err)
+	}
+
+	reg := regexp.MustCompile(`'(.*?)'`)
+
+	logStringIndex := reg.FindStringIndex(log)
+
+	if logStringIndex == nil {
+		return []string{}, nil
+	}
+
+	return cleanStringsSlice(strings.Split(log[logStringIndex[0]+1:logStringIndex[1]-1], " ")), nil
+}
+
+func (c *Client) GetListOfResourcesNs(ns, resName string) error {
+	_, err := c.kubeRunner.Get(ns, resName, "-o",
+		"jsonpath={range .items[*]}{\""+resName+" \"}\"{.metadata.name}\"{\" deleted (dry run)\"}{\"\\n\"}{end}")
+	if err != nil {
+		return fmt.Errorf("error while reading resources from cluster: %w", err)
+	}
+
+	return nil
+}
+
+func (c *Client) GetLoadBalancers() ([]string, error) {
+	log, err := c.kubeRunner.Get("all", "svc", "-o",
+		"jsonpath='{.items[?(@.spec.type==\"LoadBalancer\")].metadata.name}'")
+	if err != nil {
+		return []string{}, fmt.Errorf("error while reading resources from cluster: %w", err)
+	}
+
+	reg := regexp.MustCompile(`'(.*?)'`)
+
+	logStringIndex := reg.FindStringIndex(log)
+
+	if logStringIndex == nil {
+		return []string{}, nil
+	}
+
+	return cleanStringsSlice(strings.Split(log[logStringIndex[0]+1:logStringIndex[1]-1], " ")), nil
+}
+
+func (c *Client) DeleteAllResources(res, ns string) (string, error) {
+	result, err := c.kubeRunner.DeleteAllResources(res, ns)
+	if err != nil {
+		return result, fmt.Errorf("error while deleting resources from cluster: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Client) DeleteFromPath(path string, params ...string) (string, error) {
+	result, err := c.kubeRunner.Delete(path, params...)
+	if err != nil {
+		return result, fmt.Errorf("error while deleting resources from cluster: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Client) ToolVersion() (string, error) {
+	version, err := c.kubeRunner.Version()
+	if err != nil {
+		return "", fmt.Errorf("error while getting tool version: %w", err)
+	}
+
+	return version, nil
+}
+
+func cleanStringsSlice(slice []string) []string {
```

Comment:
I would move this to the x/slices package and make it generic, replacing the test agains the empty string with a test against the Type's zero value